core/src/parser.rs

```diff
@@ -498,6 +510,66 @@ pub(crate) fn parse_type_alias(t: &ItemType) -> Result<RustItem, ParseError> {
     }))
 }
 
+/// Parses a const variant.
+pub(crate) fn parse_const(c: &ItemConst) -> Result<RustItem, ParseError> {
+    let expr = parse_const_expr(&c.expr)?;
+
+    // serialized_as needs to be supported in case the user wants to use a different type
+    // for the constant variable in a different language
+    let ty = if let Some(ty) = get_serialized_as_type(&c.attrs) {
+        ty.parse()?
+    } else {
+        RustType::try_from(c.ty.as_ref())?
+    };
+
+    match &ty {
+        RustType::Special(SpecialRustType::HashMap(_, _))
+        | RustType::Special(SpecialRustType::Vec(_))
+        | RustType::Special(SpecialRustType::Option(_)) => {
+            return Err(ParseError::RustConstTypeInvalid);
+        }
+        RustType::Special(_s) => (),
```

Comment:
Pro-tip: simply adding this line
```
        RustType::Simple { .. } => (),
```
here would give us support already for all the type aliases to int, which could be very useful if we want to not allow all int values to be considered valid.

This requires no additional validation, as we're still validating, in `parse_const_value`, that the value attributed is an int. Therefore, the line below would allow every type that can be attributed an int.